Classes/YTPlayerView.m

```diff
@@ -509,24 +597,55 @@ + (NSString *)stringForPlayerState:(YTPlayerState)state {
       return kYTPlayerStatePausedCode;
     case kYTPlayerStateBuffering:
       return kYTPlayerStateBufferingCode;
-    case kYTPlayerStateQueued:
+    case kYTPlayerStateCued:
       return kYTPlayerStateCuedCode;
     default:
       return kYTPlayerStateUnknownCode;
   }
 }
 
+#pragma mark - WKNavigationDelegate
+
+- (void)webView:(WKWebView *)webView
+decidePolicyForNavigationAction:(WKNavigationAction *)navigationAction
+decisionHandler:(void (^)(WKNavigationActionPolicy))decisionHandler {
+  NSURLRequest *request = navigationAction.request;
+  if ([request.URL.host isEqual: self.originURL.host]) {
+    decisionHandler(WKNavigationActionPolicyAllow);
+    return;
+  } else if ([request.URL.scheme isEqual:@"ytplayer"]) {
+    [self notifyDelegateOfYouTubeCallbackUrl:request.URL];
+    decisionHandler(WKNavigationActionPolicyCancel);
+    return;
+  } else if ([request.URL.scheme isEqual: @"http"] || [request.URL.scheme isEqual:@"https"]) {
+    if ([self handleHttpNavigationToUrl:request.URL]) {
+      decisionHandler(WKNavigationActionPolicyAllow);
+    } else {
+      decisionHandler(WKNavigationActionPolicyCancel);
+    }
+    return;
+  }
+  decisionHandler(WKNavigationActionPolicyAllow);
+}
+
+- (void)webView:(WKWebView *)webView didFailNavigation:(WKNavigation *)navigation
+      withError:(NSError *)error {
+  if (self.initialLoadingView) {
+    [self.initialLoadingView removeFromSuperview];
+  }
+}
+
 #pragma mark - Private methods
 
 /**
  * Private method to handle "navigation" to a callback URL of the format
  * ytplayer://action?data=someData
- * This is how the UIWebView communicates with the containing Objective-C code.
+ * This is how the WKWebView communicates with the containing Objective-C code.
```

Comment:
Same as my comment in the header, let's just simplify to "WebView" throughout - no need to specify the exact class name (UIWebView vs. WKWebView).

Reply:
done